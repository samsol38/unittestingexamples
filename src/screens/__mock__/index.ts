const mockItems = [
  {id: 1, name: 'Item 1'},
  {id: 2, name: 'Item 2'},
  {id: 3, name: 'Item 3'},
];

const mockUserData = {
  results: [
    {
      name: {
        first: 'John',
        last: 'Doe',
      },
      email: 'john.doe@example.com',
    },
  ],
};

export {mockItems, mockUserData};
