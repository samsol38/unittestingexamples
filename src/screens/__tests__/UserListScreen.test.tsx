// Screen.test.tsx
import React from 'react';
import {render, waitFor} from '@testing-library/react-native';
import UserListScreen from '../userlist';
import fetch from 'jest-fetch-mock';

import {mockUserData} from '../__mock__';

describe('UserListScreen render with snapshot', () => {
  test('matches snapshot', () => {
    const tree = render(<UserListScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('UserListScreen', () => {
  beforeEach(() => {
    fetch.resetMocks();
    fetch.enableMocks();
  });

  test('renders loading text while fetching data', () => {
    const {getByText} = render(<UserListScreen />);
    const loadingText = getByText('Loading...');
    expect(loadingText).toBeDefined();
  });

  test('renders user data when API call is successful', async () => {
    fetch.mockResponseOnce(JSON.stringify(mockUserData));

    const {getByText} = render(<UserListScreen />);

    await waitFor(() => {
      const nameText = getByText('Name: John Doe');
      const emailText = getByText('Email: john.doe@example.com');
      expect(nameText).toBeDefined();
      expect(emailText).toBeDefined();
    });
  });

  /*
  test('handles API call error', async () => {
    fetch.mockRejectOnce(new Error('API error'));

    const {getByText} = render(<UserListScreen />);

    await waitFor(() => {
      const errorText = getByText('Error fetching user data:');
      expect(errorText).toBeDefined();
    });
  });
  */
});
