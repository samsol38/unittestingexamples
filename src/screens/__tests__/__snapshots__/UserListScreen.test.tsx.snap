// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserListScreen render with snapshot matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "paddingVertical": 50,
    }
  }
>
  <Text>
    Loading...
  </Text>
</View>
`;
